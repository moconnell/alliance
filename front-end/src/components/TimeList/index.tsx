import TimeItem from "./components/TimeItem";
import {
  Container,
  InlineContainer,
  StyledSubtitle,
  StyledTitle,
  SubText,
  TimeContainer,
} from "./styles";
import { ReactComponent as Clock } from "../../assets/clock.svg";
import { DateTime } from "luxon";

interface TimeListProps {
  times: DateTime[];
  timeZone?: string;
  selectedTime?: DateTime;
  duration?: number;
  onChange?: (time: DateTime) => void;
}

const TimeList = ({
  times,
  timeZone,
  onChange,
  selectedTime,
  duration,
}: TimeListProps) => (
  <Container>
    {times?.length && (
      <>
        <StyledTitle>Select a time</StyledTitle>

        <InlineContainer>
          <InlineContainer>
            <Clock />
          </InlineContainer>

          {duration && <StyledSubtitle>{duration} mins</StyledSubtitle>}
        </InlineContainer>

        <TimeContainer>
          {times.map((value, index) => (
            <TimeItem
              active={value === selectedTime}
              onClick={(e) => {
                e.preventDefault();
                if (onChange) onChange(value);
              }}
              key={index + "key"}
              value={value.setZone(timeZone)}
            />
          ))}
        </TimeContainer>

        <SubText>Times shown in local time</SubText>
      </>
    )}
  </Container>
);

export default TimeList;
