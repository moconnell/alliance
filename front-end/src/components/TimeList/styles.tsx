import styled from "styled-components";

export const TimeContainer = styled.div`
  width: 200px;
  max-width: 250px;
  display: flex;
  flex-direction: column;
  height: auto;
  overflow-y: auto;
  scrollbar-color: #ff3864 transparent;
  padding: 0 10px 0 10px;

  // For Google Chrome
  &::-webkit-scrollbar {
    border: 1px solid #ff3864;
    border-radius: 26px;
    width: 5px;
  }

  &::-webkit-scrollbar-thumb {
    background: #ff3864;
    border-radius: 26px;
    padding: 0.5px;
  }

  &::-webkit-scrollbar-track {
    background: transparent;
  }

  // For Internet Explorer
  & {
    scrollbar-face-color: #ff3864;
    scrollbar-track-color: transparent;
  }

  &::-moz-scrollbar {
    -moz-border: 1px solid #ff3864;
    border: 1px solid #ff3864;
    border-radius: 26px;
    width: 5px;
  }
  /* Track */

  &::-moz-scrollbar-track {
    background: transparent;
    border: 1px solid #ff3864;
    border-radius: 26px;
  }
  /* Handle */

  &::-moz-scrollbar-thumb {
    background: #ff3864;
    padding: 0.5px;
  }
`;

export const StyledSubtitle = styled.span`
  font-family: Inter;
  font-style: normal;
  font-size: 0.8em;
  line-height: 1em;
  display: flex;
  justify-content: center;
  align-items: center;
  color: #ffffff;
  margin: 0 0 0.3em 0.3em;
`;

export const SubText = styled.div`
  font-family: Inter;
  font-style: normal;
  font-size: 0.8em;
  line-height: 1em;
  display: flex;
  align-items: center;
  color: #ffffff;
  margin: 1.1em 0 0 0;
`;

export const Container = styled.div`
  padding: 0 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 400px;
  max-height: 400px;
  max-width: 250px;
`;

export const StyledTitle = styled.div`
  font-family: Inter;
  font-style: normal;
  font-weight: bold;
  font-size: 0.9em;
  display: flex;
  justify-content: center;
  align-items: center;
  color: #ffffff;
  margin: 0 0 1.4em 0;
`;

export const InlineContainer = styled.div`
  margin: 2px 8px 8px 0;
  display: flex;
  justify-content: center;
  align-items: center;
`;
