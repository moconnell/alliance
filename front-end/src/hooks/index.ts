export type { Web3Client } from "./Web3Client";
export { useWeb3 } from "./Web3Client";
export { useCalendar } from "./useCalendar";
