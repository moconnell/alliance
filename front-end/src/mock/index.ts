export const MOCK_TIMES = [
  {
    hours: 10,
    minutes: 0,
  },
  {
    hours: 10,
    minutes: 30,
  },
  {
    hours: 11,
    minutes: 0,
  },
  {
    hours: 11,
    minutes: 30,
  },
  {
    hours: 12,
    minutes: 0,
  },
  {
    hours: 13,
    minutes: 0,
  },
  {
    hours: 13,
    minutes: 30,
  },
  {
    hours: 14,
    minutes: 0,
  },
  {
    hours: 14,
    minutes: 30,
  },
  {
    hours: 15,
    minutes: 0,
  },
];

export const MOCK_MEETING = {
  attendee: "William Wallace",
  date: new Date(),
  endDate: new Date(new Date().setHours(23)),
  description:
    "Meeting to discuss the next steps in the Raid Guild World Domination Schedule ",
};

export const MOCK_MEETINGS = [
  {
    attendee: "William Wallace",
    date: new Date(),
    endDate: new Date(new Date().setHours(6)),
    description:
      "Meeting to discuss the next steps in the Raid Guild World Domination Schedule ",
  },
  {
    attendee: "Brendan Brandon",
    date: new Date(new Date().setHours(10)),
    endDate: new Date(new Date().setHours(11)),
    description: "Meeting to discuss important stuff ",
  },
  {
    attendee: "Matt Murdock",
    date: new Date(new Date().setHours(15)),
    endDate: new Date(new Date().setHours(16)),
    description: "Visual impaired Society Town Hall ",
  },
  {
    attendee: "Jhon Salvatore",
    date: new Date(new Date().setHours(23)),
    endDate: new Date(new Date().setHours(24)),
    description: "Pizza Party!",
  },
];

export const MOCK_PROFILE = {
  address: "0x309D80bd9610696b861af6fD51947e7d169bb16C",
  picture:
    "http://cdn.shopify.com/s/files/1/0583/5341/8448/products/sadPepe_digital_art_x4_8ae327af-c647-4b1d-aa50-20572ab14e13.png?v=1625845340",
  username: "sadpepe42",
  description: `Don't underestimate the Force. I find your lack of faith disturbing. Remember, a Jedi can feel the Force flowing through him. Look, I ain't in this for your revolution, and I'm not in it for you, Princess. I expect to be well paid. I'm in it for the money.`,
};
