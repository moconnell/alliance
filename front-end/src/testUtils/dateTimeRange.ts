import { DateTime } from "luxon";
import { range } from "../utils/range";

export const dateTimeRange = (
  start: number,
  end: number,
  intervalMinutes: 60 | 30 | 20 | 15 | 10 | 5 = 60,
  locale: string = "en-us"
) =>
  range(start, end).flatMap((hour) =>
    range(0, 60 / intervalMinutes - 1).map((i) =>
      DateTime.fromObject({ hour, minute: i * intervalMinutes }).setLocale(
        locale
      )
    )
  );
