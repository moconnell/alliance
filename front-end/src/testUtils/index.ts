export { dateTimeRange } from "./dateTimeRange";
export { renderWithFormProvider } from "./reactHookFormUtils";
