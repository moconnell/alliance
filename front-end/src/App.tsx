import { BrowserRouter, Route, Routes, Navigate } from "react-router-dom";
import { ErrorBoundary } from "react-error-boundary";
import "./App.css";
import Book from "./containers/Book";
import Home from "./containers/Home";
import Header from "./components/Header";
import Preferences from "./containers/Preferences";
import Meetings from "./containers/Meetings";
import ErrorFallback from "./ErrorFallback";
import { Container, Content } from "./styles";
import { useToast } from "@chakra-ui/react";

const App = () => {
  const toast = useToast();

  return (
    <main
      style={{
        background: "#333333",
        width: "vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        flexDirection: "column",
        overflowY: "scroll",
      }}
    >
      <Container>
        <Header />
        <Content>
          <ErrorBoundary
            FallbackComponent={ErrorFallback}
            onError={(e) => {
              console.error(e);
              toast({
                title: "Error",
                description: e.message,
                status: "error",
                duration: 9000,
                isClosable: true,
              });
            }}
          >
            <BrowserRouter>
              <Routes>
                <Route path="/" element={<Home />} />
                <Route path="/book" element={<Book />}>
                  <Route path="/book/:calendarAddress" element={<Book />} />
                </Route>
                <Route path="/meetings" element={<Meetings />} />
                <Route path="/profile" element={<Preferences />} />
                <Route path="*" element={<Navigate to="/" />} />
              </Routes>
            </BrowserRouter>
          </ErrorBoundary>
        </Content>
      </Container>
    </main>
  );
};

export default App;
