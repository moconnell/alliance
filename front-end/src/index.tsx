import * as React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import "./index.css";
import App from "./App";
import theme from "./theme";
import "./theme/styles.css";
import Web3ContextProvider from "./context/Web3Context";
import { createRoot } from "react-dom/client";
const container = document.getElementById("root");
const root = createRoot(container as Element);

root.render(
  <React.StrictMode>
    <ChakraProvider theme={theme}>
      <Web3ContextProvider>
        <App />
      </Web3ContextProvider>
    </ChakraProvider>
  </React.StrictMode>
);
