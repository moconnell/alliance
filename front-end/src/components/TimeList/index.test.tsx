import { fireEvent, render } from "@testing-library/react";
import { DateTime } from "luxon";
import { dateTimeRange } from "src/testUtils";
import TimeList from "./index";

describe("<TimeList />", () => {
  const mockTimes = dateTimeRange(9, 14, 30);
  const testData = [
    {},
    { selectedTime: mockTimes[0] },
    { selectedTime: mockTimes[0], duration: 30 },
  ];

  testData.forEach(({ selectedTime, duration }) =>
    it("should render correctly", async () => {
      const { asFragment } = render(
        <TimeList
          times={mockTimes}
          selectedTime={selectedTime}
          duration={duration}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    })
  );

  it("should pass value on change", async () => {
    const mockHandler = jest.fn();

    const { getByText } = render(
      <TimeList times={mockTimes} onChange={mockHandler} />
    );

    fireEvent.click(getByText("10:30 AM"));

    expect(mockHandler).toBeCalledWith(mockTimes[3]);
  });

  it("should ignore click when no onChange handler", async () => {
    const { getByText } = render(<TimeList times={mockTimes} />);

    fireEvent.click(getByText("10:30 AM"));
  });
});
