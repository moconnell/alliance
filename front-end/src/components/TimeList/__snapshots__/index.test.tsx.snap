// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeList /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-eDDNvR eUSrmv"
  >
    <div
      class="sc-jTrPJq jEgAbg"
    >
      Select a time
    </div>
    <div
      class="sc-gLDzan dslaUA"
    >
      <div
        class="sc-gLDzan dslaUA"
      >
        <svg>
          clock.svg
        </svg>
      </div>
    </div>
    <div
      class="sc-gueYoa fcbXzR"
    >
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        9:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        9:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        10:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        10:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        11:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        11:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        12:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        12:30 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        1:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        1:30 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        2:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        2:30 PM
      </button>
    </div>
    <div
      class="sc-hLseeU kLESqp"
    >
      Times shown in local time
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TimeList /> should render correctly 2`] = `
<DocumentFragment>
  <div
    class="sc-eDDNvR eUSrmv"
  >
    <div
      class="sc-jTrPJq jEgAbg"
    >
      Select a time
    </div>
    <div
      class="sc-gLDzan dslaUA"
    >
      <div
        class="sc-gLDzan dslaUA"
      >
        <svg>
          clock.svg
        </svg>
      </div>
    </div>
    <div
      class="sc-gueYoa fcbXzR"
    >
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: rgb(255, 56, 100);"
      >
        9:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        9:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        10:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        10:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        11:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        11:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        12:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        12:30 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        1:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        1:30 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        2:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        2:30 PM
      </button>
    </div>
    <div
      class="sc-hLseeU kLESqp"
    >
      Times shown in local time
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TimeList /> should render correctly 3`] = `
<DocumentFragment>
  <div
    class="sc-eDDNvR eUSrmv"
  >
    <div
      class="sc-jTrPJq jEgAbg"
    >
      Select a time
    </div>
    <div
      class="sc-gLDzan dslaUA"
    >
      <div
        class="sc-gLDzan dslaUA"
      >
        <svg>
          clock.svg
        </svg>
      </div>
      <span
        class="sc-dmqHEX glbJNG"
      >
        30 mins
      </span>
    </div>
    <div
      class="sc-gueYoa fcbXzR"
    >
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: rgb(255, 56, 100);"
      >
        9:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        9:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        10:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        10:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        11:00 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        11:30 AM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        12:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        12:30 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        1:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        1:30 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        2:00 PM
      </button>
      <button
        class="sc-beqWaB cQqxDI"
        style="background-color: transparent;"
      >
        2:30 PM
      </button>
    </div>
    <div
      class="sc-hLseeU kLESqp"
    >
      Times shown in local time
    </div>
  </div>
</DocumentFragment>
`;
